### ISSUE TITLE
User Can Accept Shares from Other Users

### ISSUE THREAD

**Issue by MTRichards** [owncloud/core#19153]
User Accept Shares: As a user, I want to confirm acceptance of files shared with me before I add them to my account so that I can ensure audit controls (based on notifications app)
Acceptance Criteria:
-   This confirmation is optional, and the overall capability is an on/off switch in the admin panel
-   When files are shared, the owners checks a box “confirm acceptance” which sends a notification and waits on an acceptance of the share
-   If the share is accepted, it appears in the recipients account. 
-   If the share is not accepted, the share owner is notified in their activity stream.

**Comment by PVince81** [comment]
@schiesbn @jancborchardt this was discussed a long time ago :smile:

**Comment by jancborchardt** [comment]
Yup, as already mentioned way back in https://github.com/owncloud/core/issues/4437

> When files are shared, the owners checks a box “confirm acceptance” which sends a notification and waits on an acceptance of the share

Not another one of these options though – at least not being the default. Shares should always need to be accepted, for reasons of storage space, unexpected files turning up in your account etc. Yes maybe we’ll have a special option or plugin which allows people to force shares, but an option like this is not needed by default.

Part of this is also having the overview page of pending & rejected shares: https://github.com/owncloud/core/issues/13495

cc @owncloud/designers

**Comment by cdamken** [comment]
@MTRichards 

00004636

**Comment by PVince81** [comment]
This could also solve some performance issues: whenever a share is created, there is code that will try and resolve file name conflicts for example if you received a folder "test" but already had a folder with this name locally. The conflict resolution can be slow when sharing with big groups.

If every user needs to accept their share separately, the resolution can be done at that time instead of at the time the share is created.

Is that actually still the case in 9.0 though ? @rullzer @schiesbn

**Comment by MTRichards** [comment]
This is a pretty big feature, but certainly one that has been out there for a long time. Interested in the state in 9.0, and also with federated sharing in 9.0, then we can discuss if we want to do this / how we want to do this.

There are plenty of use cases where this is not wanted or needed, and some where it is wanted.

**Comment by MTRichards** [comment]
I agree completely with this statement above:
>Not another one of these options though – at least not being the default. Shares should always need to be accepted, for reasons of storage space, unexpected files turning up in your account etc. Yes maybe we’ll have a special option or plugin which allows people to force shares, but an option like this is not needed by default.

We do have lots of users who are happy without this feature, and we can't change the default behavior, or clutter it up.

**Comment by rullzer** [comment]
@PVince81 it is better in 9.0... but not completely solved.

@MTRichards for now the behaviour is the same as in 8.2 et al. So federated shares need to be accepted. While user/group shares just appear. There will need to be code changes to accept all shares. But with the new sharing code in place we at least know where to look ;)

I do however agree with @jancborchardt that there needs to be 1 default. Configuring this with options is asking for trouble. I also feel more that an app could take care of this then (some hook is thrown already that could set the accepted state) or something.

So long story short. 9.0 has the same behaviour as before.

**Comment by MTRichards** [comment]
Great, thanks - we can discuss when we get around to planning for 9.1, if it makes the priority cut.

**Comment by PVince81** [comment]
This would fit well with the proposed overview page for accepted/pending/rejected shares (local and remote): https://github.com/owncloud/core/issues/13495

**Comment by butonic** [comment]
came up again

**Comment by PVince81** [comment]
@pmaier1 @felixboehm I'd like to move this to backlog for now, there is no time for this in 9.2.

Very rough estimate 10-15 md.

**Comment by pmaier1** [comment]
Probably also related to this https://github.com/owncloud/platform/issues/44.
As it is no dependency for the agreed upon items for the next release I think you can do so. @butonic what does 'came up again' mean? by whom? Anyway, this should not fall into oblivion and be discussed for the following release.

**Comment by PVince81** [comment]
Moving to backlog then.

Since this is a public ticket I'll just mention that the platform ticket above is about https://github.com/owncloud/android/issues/1676 and https://github.com/owncloud/ios/issues/656 to have mobile accept federated shares through notification apis

**Comment by PVince81** [comment]
- [ ] need pending indicator for sharer in the share list panel

**Comment by PVince81** [comment]
- [ ] auto-accept for federated shares and local shares: https://github.com/owncloud/core/issues/27653

**Comment by PVince81** [comment]
WIP PR: https://github.com/owncloud/core/pull/30106

**Comment by PVince81** [comment]
waiting for https://github.com/owncloud/core/pull/31613

**Comment by PVince81** [comment]
merged, will be released in 10.0.9
